dashboard/lti_new.py

```diff
@@ -0,0 +1,196 @@
+from django.contrib.auth.models import User
+from django.conf import settings
+from django.core.exceptions import ObjectDoesNotExist
+
+from pylti1p3.contrib.django import DjangoOIDCLogin, DjangoMessageLaunch, DjangoCacheDataStorage
+from pylti1p3.tool_config import ToolConfDict
+
+from dashboard.models import Course, CourseViewOption
+from dashboard.common.db_util import canvas_id_to_incremented_id
+
+
+logger = logging.getLogger(__name__)
+INSTRUCTOR = 'http://purl.imsglobal.org/vocab/lis/v2/membership#Instructor'
+TA = 'http://purl.imsglobal.org/vocab/lis/v2/membership/Instructor#TeachingAssistant'
+COURSE_MEMBERSHIP = 'http://purl.imsglobal.org/vocab/lis/v2/membership'
+
+
+def get_tool_conf():
+    lti_config = settings.LTI_CONFIG
+    try:
+        tool_config = ToolConfDict(lti_config)
+    except Exception as e:
+        raise e
+    # There should be one key per platform and the name relay on platforms generic domain not institution specific
+    platform_domain = list(lti_config.keys())[0]
+    client_id = lti_config[platform_domain][0]['client_id']
+    private_key_file_path = lti_config[platform_domain][0]['private_key_file']
+    public_key_file_path = lti_config[platform_domain][0]['public_key_file']
+    public_key = public_key_file_path if public_key_file_path else '/secret/public.key'
+    private_key = private_key_file_path if private_key_file_path else '/secret/private.key'
+    try:
+        private_key_content = open(private_key, 'r').read()
+        public_key_content = open(public_key, 'r').read()
+    except OSError as e:
+        raise e
+    tool_config.set_public_key(platform_domain, public_key_content, client_id=client_id)
+    tool_config.set_private_key(platform_domain, private_key_content, client_id=client_id)
+    return tool_config
+
+
+def get_jwks(request):
+    try:
+        tool_conf = get_tool_conf()
+    except (OSError, Exception) as e:
+        error_message = {
+            'lti_error_message': f'Fetching JWKS failed due to {e}'
+        }
+        return JsonResponse(error_message, status=500)
+    return JsonResponse(tool_conf.get_jwks())
+
+
+def check_if_instructor(roles, username, course_id):
+    if INSTRUCTOR in roles:
+        logger.info(f'user {username} is Instructor in the course {course_id}')
+        return True
+    return False
+
+
+def course_user_roles(roles, username):
+    user_membership_roles = set([role for role in roles if role.find(COURSE_MEMBERSHIP) == 0])
+    if not user_membership_roles:
+        logger.info(f'User {username} does not have course membership roles, must be admin {roles}')
+    elif INSTRUCTOR in user_membership_roles and TA in user_membership_roles:
+        logger.info(f'User {username} is a {TA} in the course')
+        user_membership_roles.remove(INSTRUCTOR)
+    return user_membership_roles
+
+
+def short_user_role_list(roles):
+    short_role = []
+    for role in roles:
+        short_role.append(role.split('#')[1])
+    return short_role
```

Comment:
I meant to suggest this earlier…

```suggestion
def short_user_role_list(roles):
    return [role.split('#')[1] for role in roles]
```